StackCalculator/Program.cs

```diff
@@ -0,0 +1,238 @@
+    }
+    public void Push(T value)
+    {
+        list.Add(value);
+        size++;
+    }
+    public T Pop()
+    {
+        if (size > 0)
+        {
+            T returningValue = list[^1];
+            size--;
+            list.RemoveAt(size);
+
+            return returningValue;
+        }
+        else
+        {
+            throw new InvalidOperationException("Stack in empty");
+        }
+    }
+    public int Size
+    {
+        get { return size; }
+    }
+}
+
+/// <summary>
+/// Implementation of a stack using an array.
+/// </summary>
+/// <typeparam name="T"></typeparam>
+public class StackOnArray<T> : IStack<T>
+{
+    private int size;
+    private T[] array;
+
+    /// <summary>
+    /// StackOnArray constructor.
+    /// </summary>
+    /// <param name="capacity">Maximum size of stack.</param>
+    public StackOnArray()
+    {
+        size = 0;
+        array = new T[32];
+    }
+
+    public void Push(T value)
+    {
+        if (size >= array.Length)
+        {
+            T[] tempArray = new T[array.Length * 2];
+            Array.Copy(array, tempArray, array.Length);
+            array = tempArray;
+            array[size] = value;
+        }
+
+        array[size] = value;
+        size++;
+    }
+
+    public T Pop()
+    {
+        if (size > 0)
+        {
+            size--;
+            T returningValue = array[size];
+            array[size] = default;
+            return returningValue;
+        }
+        else
+        {
+            throw new InvalidOperationException("Stack is empty");
+        }
+    }
+
+    public int Size
+    {
+        get { return size; }
+    }
+}
+
+/// <summary>
+/// Static class for performing calculations using a stack.
+/// </summary>
+public static class StackCalculator
+{
+    /// <summary>
+    /// A method for calculating the value of an expression in Polish notation.
+    /// </summary>
+    /// <param name="expression">Expression in polish notation.</param>
+    /// <param name="stackType">Type of stack to be used. Default is StackOnList</param>
+    /// <returns>Value of expression.</returns>
+    /// <exception cref="ArgumentException">If invalid input.</exception>
+    /// <exception cref="DivideByZeroException">If division by zero.</exception>
+    public static float Calculate(string expression, string stackType = "list")
+    {
+        IStack<float> stack;
+        if (stackType == "list")
+        {
+            stack = new StackOnList<float>();
+        }
+        else if (stackType == "array")
+        {
+            stack = new StackOnArray<float>();
+        }
+        else
+        {
+            throw new ArgumentException("Unsupported stack type");
+        }
+
+        List<string> elements = new(expression.Split(' '));
+        foreach (string element in elements)
+        {
+            if (int.TryParse(element, out int number))
+            {
+                stack.Push(number);
+            }
+            else
+            {
```

Comment:
```suggestion
                stack.Push(number);
                continue;
            }
```
И весь последующий код из блока `else`